frontend/e-buddy/src/app/components/FileUpload.tsx

```
@@ -0,0 +1,121 @@
+              key={file.name}
+              file={file}
+              progress={uploadProgress[file.name] || 0}
+              onRemove={handleRemoveFile}
+            />
+          ))}
+        </ul>
+        <button
+          onClick={handleUpload}
+          disabled={uploading || files.length === 0}
+          className="mt-6 w-full bg-blue-500 hover:bg-blue-600 text-white font-semibold py-3 rounded-lg transition duration-300 disabled:opacity-50"
+        >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)